### ISSUE TITLE
SignInSignUp Component

### ISSUE THREAD

**Issue by ddfridley** [EnCiv/undebate-ssp#108]
```
<SignInSignUp
    className={}
    style={}
    name={''}
    onDone={({valid, value})=>{})
>
```
onDone is a function that is called when the user is logged in. Value is undefined or userInfo returned when the user logs in, valid should be true if login successful, false if not.

- [ ] Initially, the component will just render a button, that shows the name. This will replace [SignInButton](https://github.com/EnCiv/undebate-ssp/blob/main/app/components/sign-in-button.js) and [SignUpButton](https://github.com/EnCiv/undebate-ssp/blob/main/app/components/sign-up-button.js)
- [ ] If the user clicks on the button, then the modal will pop up and lead them through the steps.
- [ ] The modal will be centered vertically and horizontally.  The width will be set in rem.  This needs to work in landscape and portrait layouts, on any resolution device.  
- [ ] If the user clicks outside of the login modal, the modal will disappear
- [ ] Update the UndebatesHeaderBar to use this component for both the Sign In Button and the Sign Up button. 
- [ ] Remove the sign-in-button.js and sign-up-button.js files 
- [ ] To implement the login functionality see [useAuth](https://github.com/EnCiv/civil-client#useauth) in the civil-client. 
Also see the [use-auth.stories.js](https://github.com/EnCiv/civil-client/blob/main/stories/use-auth.stories.js) for how this can be built and tested in storybook.  It may also be helpful to install the repo github.com/EnCiv/civil-server (in a separate directory) and run storybook there.
- [ ] to enable testomg of useAuth within storybook in this repo copy [unpoll/.storybook/middleware.js](https://github.com/EnCiv/unpoll/blob/main/.storybook/middleware.js) into this project
- [ ] The figma shows the Log In tab. If the user clicks on Sign Up, then the additional fields for confirm password and agree to the terms of service should be shown.  And the button at the bottom should change to Sign Up
- [ ] For now, we don't have a Remember Me options - so don't implement that. 
- [ ] If the name contains 'in' then start with the SignIn tab, otherwise the SignUp tab
- [ ] After getting it working in storybook, create a SignInSignUpPage component in app/web-components. See [join.js](https://github.com/EnCiv/civil-server/blob/main/app/web-components/join.js) in the civil-server repo as a starting point. 
- [ ] Also create an entry in iotas.json with the path: "/qa/signinsignuppage", see the entry in with path: "home" as a starting point.  After those are created, and after you have setup a MONGODB_URI, you will be able to `npm run dev` and then browse to localhost:3011/qa/signinsignuppage and test it with the server. 

[![image](https://user-images.githubusercontent.com/3317487/154728922-6708dc5e-1048-4989-8497-c1558956389e.png)](https://www.figma.com/file/yXrcMvg5EIwdLkHHxIdILh/Undebate-Self-Service-Portal?node-id=2268%3A5257)

**Comment by ddfridley** [comment]
@Gluay88 I think you need to do this:

- [ ] to enable testing of useAuth within storybook in this repo copy [unpoll/.storybook/middleware.js](https://github.com/EnCiv/unpoll/blob/main/.storybook/middleware.js) into this project

**Comment by ddfridley** [comment]
closed by #129
